src/vs/platform/clipboard/browser/clipboardService.ts

```diff
@@ -13,6 +13,10 @@ export class BrowserClipboardService implements IClipboardService {
 
 	private readonly mapTextToType = new Map<string, string>(); // unsupported in web (only in-memory)
 
+	async write(data: any, type?: string): Promise<void> {
```

Comment:
I don't see why this is necessary - but regardless make sure you add {{ SQL CARBON EDIT }} to all lines you modify in the vs dir

Reply:
I have added write method to IClipboardService which supports copying plain text as well as html formatted text https://www.electronjs.org/docs/api/clipboard#clipboardwritedata-type
Currently BrowserClipboardService  and NativeClipboardService both implements IClipboardService. For NativeClipboardService, I have implemented the above method whereas for BrowserClipboardService , it throws "Not Implemented" error. Another option is to only add write method in NativeClipboardService but not sure how it will affect ADS web for this command scenario.

Reply:
Gotcha - so basically for things that support HTML the clipboard will paste the HTML text otherwise it'll fall back to plain text? What's the behavior for things like Excel, Notepad, Outlook, etc?